tests/core/gui/test_password_validation.py

```diff
@@ -25,3 +26,56 @@ def test_password_text(core_config):
     assert get_password_strength_text(3) == "AVERAGE"
     assert get_password_strength_text(4) == "GOOD"
     assert get_password_strength_text(5) == "EXCELLENT"
+
+
+@pytest.mark.gui
+def test_password_validation_with_user_inputs():
+    assert get_password_strength("William J Blazkowicz") == 5
+    assert get_password_strength("William J Blazkowicz", user_inputs=["william", "blazkowicz"]) == 4
+
+
+@pytest.mark.gui
+def test_password_choice_widget(qtbot, core_config):
+    switch_language(core_config, "en")
+
+    p = PasswordChoiceWidget(parent=None)
+    qtbot.addWidget(p)
+
+    p.line_edit_password.setText("William J Blazkowicz")
+    p.line_edit_password_check.setText("William J Blazkowicz")
+
+    assert p.pwd_str_widget.label.text() == "Password strength: EXCELLENT"
+    assert p.password == "William J Blazkowicz"
+    assert p.is_valid()
+
+
+@pytest.mark.gui
+def test_password_choice_widget_mismatch(qtbot, core_config):
+    switch_language(core_config, "en")
+
+    p = PasswordChoiceWidget(parent=None)
+    qtbot.addWidget(p)
+
+    p.line_edit_password.setText("William J Blazkowicz")
+    p.line_edit_password_check.setText("William J Blazkowiz")
+
+    assert p.pwd_str_widget.label.text() == "Password strength: EXCELLENT"
+    assert p.password == "William J Blazkowicz"
+    assert not p.is_valid()
+    assert not p.label_mismatch.isHidden()
+    assert p.label_mismatch.text() == "Does not match the password."
+
+
+@pytest.mark.gui
+def test_password_choice_widget_with_user_inputs(qtbot, core_config):
+    switch_language(core_config, "en")
+
+    p = PasswordChoiceWidget(parent=None)
+    p.user_inputs = ["william.j.blazkowicz@wolfenstein.de"]
```

Comment:
should be `@wolfenstein.at`
https://en.wikipedia.org/wiki/Hohenwerfen_Castle
:trollface:

Reply:
Nop! 
https://en.wikipedia.org/wiki/Wewelsburg

Reply:
xD